File: Python/secantmethod.py
import math

x0 = input("x0: ")
x1 = input("x1: ")
eps = 1e-6
n = 0


def f(x):
    return x*math.exp(-x)+math.pow(x, 3.0)+1

print('n', " x0", "  x1", "    f(x0)", "  f(x1)", "  x2", " |Dx|")

while True:
    x2 = ((x0 * f(x1)) - (x1 * f(x0))) / (f(x1) - f(x0))
    Dx = math.fabs(x2 - x1)
    print(n+1, "%.4f" % x0, "%.4f" % x1, "%.4f" % f(x0), "%.4f" % f(x1), "%.4f" % x2, "%.4f" % Dx)
    x0 = x1
    x1 = x2
    n += 1
    if Dx < eps:
        break

File: Go/secantmethod.go
package main

import (
	"fmt"
	"math"
)

func f(x float64) float64 {
	return x*math.Exp(-x) + math.Pow(x, 3.0) + 1
}

func main() {
	n := 0
	eps := 1e-6
	var x0 float64
	var x1 float64
	fmt.Println("x0: ")
	fmt.Scanln(&x0)
	fmt.Println("x1: ")
	fmt.Scanln(&x1)

	fmt.Println("n", "x0", "x1", "f(x0)", "f(x1)", "x2", "|Dx|")
	for {
		x2 := ((x0 * f(x1)) - (x1 * f(x0))) / (f(x1) - f(x0))
		Dx := math.Abs(x2 - x1)
		fmt.Println(n+1, x0, x1, f(x0), f(x1), x2, Dx)
		x0 = x1
		x1 = x2
		n++
		if Dx < eps {
			break
		}
	}
}
